src/rgw/rgw_sal_motr.cc

```diff
@@ -3576,6 +3595,9 @@ int MotrStore::next_query_by_name(string idx_name,
     else
       next_key = key_out[i + k - 1] + " ";
     ldout(cctx, 0) << "do_idx_next_op(): next_key=" << next_key << dendl;
+    // Resizing keys & vals vector to the size of remaining keys.
+    keys.resize(keys_left);
+    vals.resize(keys_left);
```

Comment:
In this function `next_query_by_name()`, we access keys[ind] multiple times and due to IndexOutOfBound, the rgw instance used to crash, now I have resized the vectors but such errors should not crash the server according to me, 
so can wrap the function call in try-catch, but what should be the ERROR CODE I wanted to know.